crates/ra_mbe/src/mbe_expander/transcriber.rs

```diff
@@ -14,21 +14,24 @@ impl Bindings {
         self.inner.contains_key(name)
     }
 
-    fn get(&self, name: &str, nesting: &[usize]) -> Result<&Fragment, ExpandError> {
+    fn get(&self, name: &str, nesting: &mut [NestingState]) -> Result<&Fragment, ExpandError> {
         let mut b = self.inner.get(name).ok_or_else(|| {
             ExpandError::BindingError(format!("could not find binding `{}`", name))
         })?;
-        for &idx in nesting.iter() {
+        for s in nesting.iter_mut() {
+            s.hit = true;
             b = match b {
                 Binding::Fragment(_) => break,
-                Binding::Nested(bs) => bs.get(idx).ok_or_else(|| {
+                Binding::Nested(bs) => bs.get(s.idx).ok_or_else(|| {
+                    s.at_end = true;
                     ExpandError::BindingError(format!("could not find nested binding `{}`", name))
```

Comment:
So `at_end` is an error state for nested binding, right ?,  Maybe we could introduce `ExpandError::NestedError` and recover it later? Although I am okay for current approach but I just leave a record here for the alternative solution.

Reply:
Well at_end, is a marker to tell the loop it can stop now.
I did try to add a new ExpandError::AtEnd, recover from it in expand_repeat, but I hit a wall as only returning an Error (or even None) doesn't provide any information back about which nested level is at_end.

Reply:
Will `ExpandError::AtEnd(usize)`  do the trick here ? But anyway, I am okay for current approach.

Reply:
Probably. But imo it would only be worth it if the `hit` marker is also removed and replaced with a proper solution.